### ISSUE TITLE
The 'retab' command mangles non-leading tabs

### ISSUE THREAD

**Issue by ricksladkey** [jaredpar/VsVim#2493]
I came across a code base that had mixed tabs and spaces and I was trying to clean it up. I first tried to use Visual Studio Edit -> Advanced -> Untabify Selected Lines, but this turned out only to untabify leading tabs.

I thought to myself, surely vim has a builtin solution, and I shortly learned about `:retab`. While it does have the advantage of untabifying non-leading tabs, it unfortunately does not expand them correctly.

Here is a simple example. First create a text buffer using the settings `:set noexpandtab tabstop=4`:

```
cat<Tab>dog
```
which will be rendered as:

```
cat dog
```
Now set `:set expandtab` and run `:retab` and the result should not visually change, but the buffer now looks like this:

```
cat    dog
```

In other words, one tab (that started on column 3 and took up one screen column) was expanded to four spaces, as though the tab were aligned to a multiple of four.
